galley/pkg/config/analysis/msg/messages.yaml

```diff
@@ -47,3 +47,25 @@ messages:
         type: string
       - name: refval
         type: string
+
+  - name: "NamespaceNotInjected"
+    code: IST0102
+    level: Warning
+    description: "A namespace is not enabled for Istio injection."
+    template: "The namespace is not enabled for Istio injection. Run 'kubectl label namespace %s istio-injection=enabled' to enable it, or 'kubectl label namespace %s istio-injection=disabled' to explicitly mark it as not needing injection"
```

Comment:
I think we should make a distinction between a short description and a long description in the toolchain. I think this prose might be a bit long for putting in a command-output message.

We can also have a URL associated with each message, so that more detailed documentation and remediation actions can be found online.

Reply:
I'm not even seeing `description` being used anywhere. @sushicw?

Ideally, I'd like to give direct actionable items that users can cut and paste, rather than send them to docs.

But if we need to go down the doc path, we'll need to create redirection links so that we can point them to the right place. Do you think it would be good enough to point them to docs on the Istio wiki? We need to be able to update things efficiently and without too much process.

Reply:
Description is not currently used, I originally introduced it as a simple way to capture more detailed information, so that we can generate online collateral for documentation. 

In my opinion, we should have a page/sub-site in istio.io categorized by the error code, and they can have Istio version specific sub-sections for each Istio version. That way we can generate a URL directly from the error code: e.g.: `http://istio.io/docs/reference/codes/ist0101/#istio1.3` (I am sure we can do better than this, I am giving this as an example).

So, given an error code, it should be trivial to generate a URL link to an online doc, where we can prescribe various different solutions (or further diagnostic actions), possibly in different ways for different Istio versions.

Reply:
@geeknoid FYI.

Reply:
Yes, I like that idea.

Reply:
I'll keep the long message for now and move that to docs once we're set up for it.

Reply:
Though I prefer to keep the Istio version off the URL. A given error code should mean the same for all versions. If semantics change, we can some up with a new error code.

Reply:
The nice thing about using the template is that we're generating exactly what command the user needs to run, and they can just copy-paste it. On the flipside, this makes us a bit brittle if that mechanism ever changes. e.g. what if they aren't using kubectl at all? Remember that this message isn't just for istioctl & the command line.

I definitely agree that we should be generating & linking to online documents. This deserves some more structured thought.

Reply:
I'm also a big fan of directly giving commands that can be copied/pasted. The question is whether the tool can properly deal with messages that are long.

As for pointing to kubectl, I think that's quite acceptable for any k8s user. A lot of this is about being helpful in the common cases.